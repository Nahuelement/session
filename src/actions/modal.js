import { types } from "../types/types"




export const openModal =() =>({
    type : types.activeModal,
    
})
export const endModal =() =>({
    type : types.endModal,
    
})