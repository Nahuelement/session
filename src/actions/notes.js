import { db } from "../firebase/fireConfig";
import { types } from "../types/types";
import { collection ,addDoc, doc, setDoc, deleteDoc} from "firebase/firestore";
import { loadNotes } from "../helpers/loadNotes";
import Swal from "sweetalert2";
import { fileUpload } from "../helpers/fileUpload";

// import cloudinary from 'cloudinary'

// // react-journal

//  cloudinary.config({ 
//     cloud_name: 'nahuelement', 
//     api_key: '723161997253356', 
//     api_secret: 'c2WpIaroeOWZNhdCdN8S0CQp-WE',
//     secure: true
//  });

export const startNewNotes = () =>{

    return async(dispatch, getState) =>{ // getState para obtener el estado

        const {uid} = getState().auth;
        

        const newNote = {
            title: '', 
            body: '',
            date: new Date().getTime(),
            url: ''
        }
        try{
            const docRef = await addDoc(collection(db,`${uid}/journal/notes`),newNote)
            dispatch(activeNote(docRef.id,newNote))
            // dispatch(startLodingNotes(uid)) IDEA MIA 
            dispatch(addNewNote(docRef.id,newNote))

        }
        catch (err) {
            console.log(err)
        } 
    }
}

export const activeNote = (id, note) => ({

    type:types.notesActive,
    payload:{
        id,
        ...note
    }
})

export const addNewNote = (id,note) => ({

    type:types.notesAddNew,
    payload:{
        id,
        ...note
    }
})


export const startLodingNotes = (uid) => {

    return async(dispatch) =>{

        try{
            const notes = await loadNotes(uid)
            dispatch(setNotes(notes))
        }
        catch(err) {
            console.log(err)

        }        
    }
}



export const setNotes = (notes) => ({

    type:types.notesLoad,
    payload:notes
    
}) 

export const startSaveNote = (note) => { 

    return async (dispatch, getState) =>{

        const {uid} = getState().auth; 
        
         if(!note.url){
             delete note.url;
         } 

        const noteToFirestore = {...note};
        delete noteToFirestore.id
        // console.log(noteToFirestore)
        
        try{
            await setDoc(doc(db,`${uid}/journal/notes/${note.id}`),noteToFirestore)
            Swal.fire('Guardado!',note.title,'success')
            dispatch(refreshNote(note.id,{...note}))
        }
        catch(err) {
            console.log(err)
        }
        
        


        


       
            
        
            
       
       
    }   

}
export const refreshNote = (id,note) =>({
    type:types.notesUpdated,
    payload:{
        id,
        note
    }

})

export const startUpLoading = (file) =>{
    return async(dispatch, getState) =>{

        const {active: activeNote} = getState().notes;
        
        Swal.fire({ 
            title: 'Uploading...',
            text:' Plase wait...',
            allowOutsideClick:false,
            allowEnterKey: () =>{
                Swal.showLoading();
            }
        }) 

        let fileUrl  = await fileUpload(file)
        
        Swal.close()
         if(!fileUrl){
         fileUrl= 'www.koi.partedelaprueba.porproblemasconJest().com'
        }
        activeNote.url = fileUrl
        console.log(activeNote)
       

        dispatch(startSaveNote(activeNote))
        
    }
}

export const startDeleting = (id) => {
    return async(dispatch, getState) =>{

        const uid = getState().auth.uid

        try{
            await deleteDoc(doc(db,`${uid}/journal/notes/${id}`)) 
            dispatch(deleteNotes(id))
        }
        catch(err) {
            console.log(err)
            
        }
        

        
    }
}

export const deleteNotes = (id) => ({
    type:types.notesDelete,
    payload:id
})

export const noteLogout = () => ({
    type:types.notesLogoutCleaning,
})