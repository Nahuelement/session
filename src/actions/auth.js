import { googleAuthProvider,facebookAuthProvider} from "../firebase/fireConfig"
import { getAuth, signInWithPopup, GoogleAuthProvider ,signInWithEmailAndPassword,
         FacebookAuthProvider,createUserWithEmailAndPassword,updateProfile,
         signOut} from "firebase/auth";
import { types } from "../types/types"
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from "sweetalert2";
import { noteLogout } from "./notes";




export const startLogin = (email, password) => {

    return(dispatch) =>{
        dispatch(uiStartLoading())


        signInWithEmailAndPassword(getAuth(), email, password)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName));
            dispatch(uiFinishLoading())
       
        }).catch(e => {
            
            // const errorCode = e.code;
            const errorMessage = e.message;

            let message = errorMessage;

            if(String(errorMessage)==='Firebase: Error (auth/invalid-email).'){
                 message = 'ingrese email valido'

            }else if (String(errorMessage)==='Firebase: Error (auth/wrong-password).'){
                 message = 'Password incorrecto'
                 
            }else if (String(errorMessage)==='Firebase: Error (auth/internal-error).'){
                 message = 'Ingrese email y contraseÃ±a'

                 
            }else if (String(errorMessage)==='Firebase: Error (auth/user-not-found).'){
                message = 'Email no registrado'
                
           }
           
            dispatch(uiFinishLoading())

            Swal.fire('Error',message,'error')
            
      })

    
}
}

export const startRegisterWhitEmailPassword  = (email, password,name) => {

    return(dispatch) =>{

    createUserWithEmailAndPassword(getAuth(),email, password)
    .then(async({user}) =>{
        
        await updateProfile(getAuth().currentUser,{displayName:name})
        
        dispatch(login(user.uid,user.displayName))
       
        
    })
    .catch((error) => {
        // const errorCode = error.code;
        const errorMessage = error.message;

        let message;
        if(String(errorMessage)==='Firebase: Error (auth/email-already-in-use).'){
            message = 'Usuario ya registrado con este email'

        }
       
        Swal.fire('Error',message,'error')

    }
    )
}}

export const starGoogleLogin = () => {
    
    return(dispatch) => {
        
       
        signInWithPopup(getAuth(),googleAuthProvider)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName))
            
        }).catch((e) =>{
            
            GoogleAuthProvider.credentialFromError(e);
            
        })
        
    
    }}

    export const starFacebookLogin = () => {
    
        return(dispatch) => {
           
            signInWithPopup(getAuth(),facebookAuthProvider)
            .then((result) =>{
                const user = result.user;

                console.log(user)
            }).catch(FacebookAuthProvider.credentialFromError)
        }}
            

  

export const login = (uid, displayName) => ({

   
        type: types.login,
        payload: {
            uid,
            displayName
        }
})
export const startLogout = () => {

   return async(dispatch)=> {

    await signOut(getAuth())
    dispatch(logout())
    dispatch(noteLogout())

   }
    
}

export const logout = () => ({

    type: types.logout,
})
