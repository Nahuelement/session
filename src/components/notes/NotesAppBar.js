import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote, startUpLoading } from '../../actions/notes'




export const NotesAppBar = () => {


    const {active}= useSelector(state => state.notes)

    const {date} = active

    const dispatch = useDispatch()


    const handledSave = () => {
        console.log('Archivo activo guardado:',active)
        dispatch(startSaveNote(active))
    }

    const handleFileChange = (e) => {
        const file = e.target.files[0]
        console.log(file)
        if (file){

            dispatch(startUpLoading(file))

        }
    }

    const handledUpPicture = () => {
        document.querySelector('#fileSelector').click()
    }

    return (
        <div className="notes__appbar">

            <span> NÂª Cliente {date} </span>

            <input 
            id = 'fileSelector'
            type="file" 
            style ={{display:'none'}}
            name = 'file'
            onChange={handleFileChange}
            />
            <div>
                <button 
                className="btn"
                onClick = {handledUpPicture}
                >
                    Cargar archivo 
                </button>
                {/* <button 
                className="btn"
                onClick={handledSave}
                >
                    Guardar
                </button> */}

            </div>
        </div>
    )
}
