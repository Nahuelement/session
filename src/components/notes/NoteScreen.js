import React, { useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { NotesAppBar } from './NotesAppBar'

import {useForm} from '../../hooks/useForm'
import { useEffect } from 'react'
import { activeNote, startDeleting } from '../../actions/notes'



export const NoteScreen = () => {

    const dispatch = useDispatch();

    const { active:note } = useSelector( state => state.notes );
    const [ formValues, handleInputChange, reset ] = useForm( note );
    const { body, title, id } = formValues;

    const activeN = useRef( note );

    useEffect(() => {
        
        if ( note.id !== activeN.current.id ) {
            reset( note );
            activeN.current.id = note.id
        }else if (note.url !== activeN.current.url){
            reset( note )
            activeN.current.url = note.url
        }

    }, [note, reset])

    useEffect(() => {
        
        dispatch( activeNote( formValues.id, { ...formValues } ) );

    }, [formValues, dispatch])


    const handleDelete = () => {
        dispatch( startDeleting( id ) );
    }
    return (
        <div className="notes__main-content    ">
            
            <NotesAppBar  />

            <div className='notes__content   '>
            
                {/* <input 
                
                type="text"
                placeholder="Agrega un buen titulo"
                className ='notes__title-input animate__animated animate__fadeIn '
                onChange={handleInputChange}
                name = 'title'
                value = {title}
                
                />

                <textarea
                placeholder="Â¿Que a pasado hoy?"
                className='notes__textarea animate__animated animate__fadeIn '
                value = {body}
                name = 'body'
                onChange={handleInputChange} >

                </textarea> */}

                { note.url  && (<div className = 'notes__image animate__animated animate__fadeIn'>
                    <img 
                    src={note.url }
                    alt= 'Imagen X'
                        
                        />

                </div>)}
            </div>
            {/* <button

            className = 'btn btn-danger animate__animated animate__fadeIn animate__fast	800ms'
            onClick={handleDelete}
            >
                Eliminar
            </button> */}
            
        </div>



    )
}
