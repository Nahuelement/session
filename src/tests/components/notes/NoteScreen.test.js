import { NoteScreen } from "../../../components/notes/NoteScreen";

import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store' 
import thunk from "redux-thunk"
import {mount} from 'enzyme'
import { activeNote, startDeleting } from "../../../actions/notes";


jest.mock('../../../actions/notes',() => ({ //mock devuelve un objeto ()
    startDeleting:jest.fn(),
    activeNote: jest.fn()
}))


const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = { // inicial estado de mock que remplaza al store de redux
    auth:{},
    notes:{
        active:{
            id:'ImTMfs5frDggDoYFXhnm',
            title:'hola',
            body:'body',
            date: 43443243244234
        }
    },
    ui:{
        loading:false,
        msgError:null 
    }
}
let store =mockStore(initialState)//STORE NO UTILIZA LOS REDUCER
store.dispatch = jest.fn()

describe('Pruebas en el <NoteScreen />', () => {
    const wrapper = mount(
    
        <Provider store = {store}>
                <NoteScreen />
        </Provider>
    )

    test('debe de mostrar el NoteScreen ', () => {

        expect(wrapper).toMatchSnapshot()
        
    })
    test('debe disparar la accion  startDeleting ', () => {

        wrapper.find('.btn-danger').prop('onClick')() 
        expect(startDeleting).toBeCalled()
         
    })
    test('debe de disparar activeNote apenas cambie el estado del formulario ', () => {

        wrapper.find('input[name="title"]').simulate('change',
        {target:{
            value:'Hola de nuevo',//VALOR VACIO
            name: 'title',
        }} 
            ) 
        expect(activeNote).toHaveBeenLastCalledWith(
            "ImTMfs5frDggDoYFXhnm", 
            {
                "body": "body",
                "date": 43443243244234,
                "id": "ImTMfs5frDggDoYFXhnm",
                 "title": "Hola de nuevo"
                }
        )
        
         
    })
    
    
})
