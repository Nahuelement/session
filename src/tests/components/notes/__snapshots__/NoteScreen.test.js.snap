// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <NoteScreen /> debe de mostrar el NoteScreen  1`] = `
<div
  className="notes__main-content animate__animated  animate__fadeIn animate__fast 800ms "
>
  <div
    className="notes__appbar"
  >
    <span>
      43443243244234
       
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Guardar
      </button>
    </div>
  </div>
  <div
    className="notes__content animate__animated animate__fadeIn "
  >
    <input
      className="notes__title-input animate__animated animate__fadeIn "
      name="title"
      onChange={[Function]}
      placeholder="Agrega un buen titulo"
      type="text"
      value="hola"
    />
    <textarea
      className="notes__textarea animate__animated animate__fadeIn "
      name="body"
      onChange={[Function]}
      placeholder="Â¿Que a pasado hoy?"
      value="body"
    />
  </div>
  <button
    className="btn btn-danger animate__animated animate__fadeIn animate__fast	800ms"
    onClick={[Function]}
  >
    Eliminar
  </button>
</div>
`;
